nat-lab/tests/utils/process/docker_process.py

```diff
@@ -158,3 +161,30 @@ def is_executing(self) -> bool:
 
     async def is_done(self) -> None:
         await self._is_done.wait()
+
+
+def kill_docker_process_by_host_pid(pid: str, container_id: str):
+    """This function may look a bit weird, because we could just kill the process
+    using its PID on host, but, unfortunately, such kill is often denied due to some
+    security measures (even while using sudo)"""
+
+    # Currenlty the gentle way of killing the docker process is supported only on linux,
+    # for the other platforms we fallback to the old solution
+    if sys.platform not in ("linux", "linux2"):
+        os.system(f"sudo kill -9 {pid}")
+        return
+
+    # It may happen that the process already exited at this point, in which situation
+    # we may expect an error
+    try:
+        docker_pid = (
+            subprocess.check_output(f"cat /proc/{pid}/status | grep NSpid", shell=True)
```

Comment:
I'd like to see it implemented in Python itself, meaning reading and inspecting the `/proc/<pid>/status` file, because it adds a hidden dependency for `cat` and `grep` being installed on the host machine